src/compiler/transformers/classFields.ts

```diff
@@ -7,91 +7,38 @@ namespace ts {
          */
         ClassAliases = 1 << 0,
     }
-
-    const enum PrivateIdentifierPlacement {
-        InstanceField,
-        InstanceMethod,
-        InstanceGetterOnly,
-        InstanceSetterOnly,
-        InstanceGetterAndSetter,
-        StaticField,
-        StaticMethod,
-        StaticGetterOnly,
-        StaticSetterOnly,
-        StaticGetterAndSetter,
-    }
-
-    type PrivateIdentifierInfo =
-        | PrivateIdentifierStaticField
-        | PrivateIdentifierStaticMethod
-        | PrivateIdentifierStaticGetterOnly
-        | PrivateIdentifierStaticSetterOnly
-        | PrivateIdentifierStaticGetterAndSetterOnly
-        | PrivateIdentifierInstanceField
-        | PrivateIdentifierInstanceMethod
-        | PrivateIdentifierInstanceGetterOnly
-        | PrivateIdentifierInstanceSetterOnly
-        | PrivateIdentifierInstanceGetterAndSetter;
-
-    interface PrivateIdentifierStaticField {
-        placement: PrivateIdentifierPlacement.StaticField;
-        variableName: Identifier;
-        classConstructor: Identifier;
-    }
-    interface PrivateIdentifierStaticMethod {
-        placement: PrivateIdentifierPlacement.StaticMethod;
-        functionName: Identifier;
-        classConstructor: Identifier;
+    interface PrivateIdentifierInfoBase {
+        state: Identifier;

```

Comment:
A doc string explaining the three types of identifiers for this property would be nice.